express-api/src/services/buildings/buildingServices.ts

```diff
@@ -0,0 +1,121 @@
+import { Building } from '@/typeorm/Entities/Building';
+import { AppDataSource } from '@/appDataSource';
+import { ErrorWithCode } from '@/utilities/customErrors/ErrorWithCode';
+import { DeepPartial, FindOptionsOrder } from 'typeorm';
+import { BuildingFilter } from '@/services/buildings/buildingSchema';
+
+const buildingRepo = AppDataSource.getRepository(Building);
+
+/**
+ * @description          Adds a new building to the datasource.
+ * @param   building     Incoming building data to be added to the database.
+ * @returns {Response}   A 201 status and the building data added.
+ * @throws ErrorWithCode If the building already exists or is unable to be added.
+ */
+export const addBuilding = async (building: Building) => {
```

Comment:
This incoming building won't have all fields, like id, so we need a partial.
Either that or we define a new interface for what adding a building should take.
```suggestion
export const addBuilding = async (building: DeepPartial<Building>) => {
```